controllers/pkg/reconcilers/bootstrap-packages/reconciler.go

```diff
@@ -89,144 +82,135 @@ func (r *reconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, c a
-								return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
-							}
-						}
-						// install resources
-						for _, resource := range resources {
-							resource := resource // required to prevent gosec warning: G601 (CWE-118): Implicit memory aliasing in for loop
-							r.l.Info("install manifest", "resource",
-								fmt.Sprintf("%s.%s.%s", resource.GetAPIVersion(), resource.GetKind(), resource.GetName()))
-							if err := clusterClient.Apply(ctx, &resource); err != nil {
-								msg := fmt.Sprintf("cannot apply resource to cluster: resourceName: %s", resource.GetName())
-								r.l.Error(err, msg)
-								return ctrl.Result{}, errors.Wrap(err, msg)
-							}
-						}
+			clusterClient, ok := cluster.Cluster{Client: r.Client}.GetClusterClient(&clusterSecret)
+			if ok {
+				clusterClient, ready, err := clusterClient.GetClusterClient(ctx)
+				if err != nil {
+					msg := "cannot get clusterClient"
+					r.log.Error(err, msg)
+					return ctrl.Result{RequeueAfter: 30 * time.Second}, errors.Wrap(err, msg)
+				}
+				if !ready {
+					r.log.Info("cluster not ready")
+					return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
+				}
+				// install the resources to the cluster
+				for _, resource := range resources {
+					resource := resource // required to prevent gosec warning: G601 (CWE-118): Implicit memory aliasing in for loop
+					r.log.Info("install manifest", "resource", fmt.Sprintf("%s.%s.%s", resource.GetAPIVersion(), resource.GetKind(), resource.GetName()))
+					if err := clusterClient.Apply(ctx, &resource); err != nil {
+						msg := fmt.Sprintf("cannot apply resource to cluster: resourceName: %s", resource.GetName())
+						r.log.Error(err, msg)
+						return ctrl.Result{}, errors.Wrap(err, msg)
 					}
 				}
-			}
-			if !found {
-				// the cluster client was not found, we retry
-				r.l.Info("cluster client not found, retry...")
+			} else {
+				// the clusterClient was not found, we retry
+				r.log.Info("cluster client not found, retry...")
 				return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
 			}
 		}
 	}
 	return ctrl.Result{}, nil
 }
 
-func (r *reconciler) IsStagingPackageRevision(ctx context.Context, cr *porchv1alpha1.PackageRevision) (bool, error) {
-	repos := &porchconfigv1alpha1.RepositoryList{}
-	if err := r.porchClient.List(ctx, repos); err != nil {
+func (r *reconciler) GetClusterSecret(ctx context.Context, client client.Client, clusterName string) (corev1.Secret, error) {
+	// we need to find the cluster client secret
+	secrets := &corev1.SecretList{}
+	if err := client.List(ctx, secrets); err != nil {
+		return corev1.Secret{}, err
+	}
+	clusterSecret := corev1.Secret{}
+	for _, secret := range secrets.Items {
+		if strings.Contains(secret.GetName(), clusterName) {
+			clusterSecret = secret
+			break
+		}
+	}
+	return clusterSecret, nil
+}
 
+func (r *reconciler) IsStagingPackageRevision(ctx context.Context, porchClient client.Client, repositoryName string) (bool, error) {
```

Comment:
don't put porchClient client.Client here

Reply:
Same issue with passing the mock client.

Reply:
see above